GREEDY_WEB/ClientApp/components/FetchData.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import 'isomorphic-fetch';
+import axios from 'axios';
+import { Button, ButtonGroup, InputGroup, InputGroupAddon, Input, Form, FormGroup, Label, FormText } from 'reactstrap';
+import { ModalContainer, ModalDialog } from 'react-modal-dialog';
+
+interface Props {
+    itemList: any
+}
+
+interface State {
+    showEdit: boolean
+    showItems: boolean
+    itemList: any
+    //e - edit
+    eItemId: number
+    eItemName: string
+    eItemPrice: number
+    eItemCategory: string
+    eHappened: boolean
+    eSuccess: boolean
+}
+
+export class FetchData extends React.Component<Props, State> {
+    constructor() {
+        super();
+        /*axios.get('http://localhost:6967/api/ItemData')
+            .then(res => {
+                const data = res.data;
+                this.setState({ itemList: data, showItems: true });
+            });*/
+        this.editItem = this.editItem.bind(this);
+        this.getItems = this.getItems.bind(this);
+        this.hideEdit = this.hideEdit.bind(this);
+        this.saveItemChanges = this.saveItemChanges.bind(this);
+        this.eCategoryChange = this.eCategoryChange.bind(this);
+        this.eNameChange = this.eNameChange.bind(this);
+        this.ePriceChange = this.ePriceChange.bind(this);
```

Comment:
You can remove all this line and change methods to property like 
    update = () => {
        this.getItems();
    }